### PR TITLE
🚧 - Provide support for KEDA 2.0

### PR DESCRIPTION
**Pull request by tomkerkhove**
Provide support for KEDA 2.0.

- It will automatically determine the current version of KEDA and take according actions, otherwise it will go with 2.0.
- All actions support passing KEDA version if need be
- Using Kubernetes resource to delete required resources rather than keeping a list of resource names

Relates to #2144

### CODE REVIEW

**Comment by tomkerkhove** [comment]
I'll need some help on following topics:
- Is there a full list of trigger types that I can lookup so that I can implement `GetKedaTrigger` accordingly to map to the KEDA scalers?
- `PopulateMetadataDictionary` is to provide the metadata for scalers but I think it uses ENV VAR references by default, is that correct? Just to make sure I update the breaking changes
- Is there a way to run this locally? Based on the contributing file there are no tests but I'd like to give `KedaHelper.DetermineCurrentVersion` a go.

I don't know who will be able to help with my questions here, maybe @ahmelsayed @fabiocav or @jeffhollan?

File: src/Azure.Functions.Cli/Kubernetes/KEDA/KedaHelper.cs [hunk 1]
```diff
@@ -41,10 +43,20 @@ internal static IKubernetesResource GetScaledObject(string name, string @namespa
...
         {
-            // TODO: discover version
-            return KedaVersion.v2;
+            // Get KEDA resource information
+            var resourceInfo = await KubernetesHelper.ResourceExists("Deployment", "keda-metrics-apiserver", @namespace, returnJsonOutput: true);
+            if (resourceInfo.ResourceExists == false)
+            {
+                // We use KEDA v2 as a default
+                return KedaVersion.v2;
+            }
```
Comment (prqgdbc) [inline src/Azure.Functions.Cli/Kubernetes/KEDA/KedaHelper.cs:54]: Perhaps checking for the existence of the apiservice here instead might be more compatible with different methods of installing helm?

```suggestion
            var resourceInfoV2 = await KubernetesHelper.ResourceExists("apiservice", "v1alpha1.keda.sh", @namespace, returnJsonOutput: true);
            
            if(resourceInfoV2.ResourceExists)
            {
                return KedaVersion.v2;
            }
```

Comment (tomkerkhove) [reply src/Azure.Functions.Cli/Kubernetes/KEDA/KedaHelper.cs:54]: This way you rely on the resource type which is different between v1 & v2 which is not the case.

The ways to install KEDA all provide deployment with that name so think we are good to go here.

**Comment by tomkerkhove** [comment]
Any thoughts on this @ahmelsayed @jeffhollan?

**Comment by jeffhollan** [comment]
I have this flagged as an item to have a dev assigned for this sprint - planning is on Wednesday. If you don’t hear back by then that may be when @fabiocav assigns someone to lock and merge. Thank you a ton for doing this

**Comment by fabiocav** [comment]
Thank you, @tomkerkhove !

Assigning this to be looked at in the current sprint and we'll have folks assigned and following up with you.

**Comment by tomkerkhove** [comment]
Thanks @fabiocav !

**Comment by fabiocav** [comment]
@tomkerkhove I'm also assuming this is for 3.0, correct? If so, we need to target the v3.x branch. We'll actually be changing the default branch target this week

**Comment by tomkerkhove** [comment]
It is backwards compatible but happy to switch to v3 if you prefer?

**Comment by tomkerkhove** [comment]
I'll clean things up and switch to v3

**Comment by tomkerkhove** [comment]
There are too many changes, I'll just create a new branch based off of v3 to make sure I don't mess things up

**Comment by tomkerkhove** [comment]
Closing in favor of https://github.com/Azure/azure-functions-core-tools/pull/2289 which is targeting 3.X

**Pull request closed without merging**